woocommerce/woocommerce-ios
# [Onboarding Deep Link] Add login deep link track events

Closes: #10843 

### Why
We are improving the app onboarding experience by adding a QR that shares the site URL and the email from the wp-admin. This way, merchants don't need to enter this information when logging into the apps. This PR covers the remaining part, covering the deep link analytics.

### How
This PR adds the success and failure analytics tracks events when reading the app login deep link. 

⚠️ Event registration is pending, but I'm sorting that out.

### Testing instructions
TC1
1. Log out from the app
2. Open the terminal and run this script using your site URL and wp.com email:

```
xcrun simctl openurl booted "woocommerce://app-login?siteUrl=https://[SiteURL]&username=[Username]"
```

3. Check that the app opens on the login screen with your username
4. Check that the success track event is sent with the flow no-wp

TC2.
1. Repeat the steps above but use this WP Com link

```
xcrun simctl openurl booted "woocommerce://app-login?siteUrl=https://[SiteURL]&wpcomEmail=[WPComEmail]"
```
2. Check that the success track event is sent with the flow wp

TC3.
1. Repeat the steps above but use this malformed link

```
xcrun simctl openurl booted "woocommerce://app-login?siteUrl=https://[SiteURL]&wpcomEmail=[WPComEmail]"
```
2. Check that the malformed track event is sent with the malformed url parameter

### Images/gif


#### WP Com flow
```
🔵 Tracked: login_app_login_link_success, Properties: {"flow":"wp_com","is_debug":true}
```
#### No WP Com flow
```
🔵 Tracked: login_app_login_link_success, Properties: {"flow":"no_wp_com","is_debug":true}
```
#### Malformed URL
```
🔵 Tracked: login_malformed_app_login_link, Properties: {"url":"woocommerce:\/\/app-login?siteUrl=[SiteURL]","is_debug":true}
```
